migrations/versions/4ff089373353_added_not_null_constraint_to_study_.py

```diff
@@ -0,0 +1,77 @@
+"""Added not null constraint to study_version.status and eligibility_criteria.status.
+
+Revision ID: 4ff089373353
+Revises: 241619ddac70
+Create Date: 2024-11-07 12:46:42.837498
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = '4ff089373353'
+down_revision = '241619ddac70'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 1, study_algorithm_engine_id = 1 where id = 1")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 2, study_algorithm_engine_id = 2 where id = 2")
+    op.execute("UPDATE study_version SET status = 'INACTIVE', eligibility_criteria_id = 3, study_algorithm_engine_id = 3 where id = 3")
+    op.execute("UPDATE study_version SET status = 'INACTIVE' where id = 4")
+    op.execute("UPDATE study_version SET status = 'INACTIVE', eligibility_criteria_id = 5, study_algorithm_engine_id = 5 where id = 5")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 6, study_algorithm_engine_id = 6 where id = 6")
+    op.execute("UPDATE study_version SET status = 'INACTIVE', eligibility_criteria_id = 7, study_algorithm_engine_id = 7 where id = 7")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 8, study_algorithm_engine_id = 8 where id = 8")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 9, study_algorithm_engine_id = 9 where id = 9")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 10, study_algorithm_engine_id = 10 where id = 10")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 11, study_algorithm_engine_id = 11 where id = 11")
+    op.execute("UPDATE study_version SET status = 'INACTIVE', eligibility_criteria_id = 12, study_algorithm_engine_id = 12 where id = 12")
+    op.execute("UPDATE study_version SET status = 'INACTIVE', eligibility_criteria_id = 13, study_algorithm_engine_id = 13 where id = 13")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 14, study_algorithm_engine_id = 14 where id = 14")
+    op.execute("UPDATE study_version SET status = 'INACTIVE', eligibility_criteria_id = 15, study_algorithm_engine_id = 15 where id = 15")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 16, study_algorithm_engine_id = 16 where id = 16")
+    op.execute("UPDATE study_version SET status = 'ACTIVE', eligibility_criteria_id = 17, study_algorithm_engine_id = 17 where id = 17")
+    op.execute("UPDATE criterion set active = true")
+    op.execute("UPDATE eligibility_criteria set status = 'ACTIVE'")
```

Comment:
The IDs are different in different environments. Prod will likely have different IDs, or if we didn't load anything in prod it doesn't really matter. We should make the update based on the NCT ID or something that is the same in all the environments.

Again, if the prod DB is empty we can skip this part, so worth checking if we have any data in there before making modifications.

Reply:
This should be ready to go now.

Reply:
I forgot we have sql file with stable IDs. Thank you